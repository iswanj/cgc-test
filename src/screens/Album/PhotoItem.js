import React from 'react';
import {StyleSheet, Image} from 'react-native';
import {Colors, View, ListItem, Text} from 'react-native-ui-lib';

import {useNavigation} from '@react-navigation/native';

export const PhotoItem = ({item}) => {
  const navigation = useNavigation();

  const handleClick = id => {
    navigation.navigate('PhotoScreen', {
      item,
    });
  };
  return (
    <ListItem
      containerStyle={styles.container}
      activeBackgroundColor={Colors.grey60}
      height={150}
      width={150}
      onPress={() => handleClick(item.id)}>
      <View flex paddingH-10 centerV row>
        <View width={150} height={150}>
          <Image style={styles.image} source={{uri: item.thumbnailUrl}} />
        </View>
        <View paddingH-10 flex-5>
          <Text capital>{item.title}</Text>
        </View>
      </View>
    </ListItem>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginBottom: 10,
  },
  image: {
    width: 150,
    height: 150,
  },
});
