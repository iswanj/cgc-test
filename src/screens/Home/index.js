import React from 'react';
import {FlatList} from 'react-native';

import {View, Text, ListItem} from 'react-native-ui-lib';

import {AlbumItem} from './AlbumItem';
import {useAlbums} from '@cgc/services/apis';

export const Home = () => {
  const {albums, isLoading} = useAlbums();
  const keyExtractor = item => item.id;
  return (
    <View flex>
      <FlatList
        data={albums}
        renderItem={({item}) => <AlbumItem key={item.id} item={item} />}
        keyExtractor={keyExtractor}
      />
    </View>
  );
};
