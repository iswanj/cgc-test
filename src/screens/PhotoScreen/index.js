import React from 'react';
import {Image, StyleSheet} from 'react-native';

import {View, Text, ListItem} from 'react-native-ui-lib';

import {useRoute} from '@react-navigation/native';

export const PhotoScreen = () => {
  const route = useRoute();
  const item = route.params.item;
  return (
    <View flex centerH marginT-20>
      <View centerH style={styles.imageContainer}>
        <Image style={styles.image} source={{uri: item.url}} />
        <Text title capital paddingH-20 marginT-20>
          {item.title}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  imageContainer: {
    flex: 1,
  },
  image: {
    width: 300,
    height: 300,
  },
});
