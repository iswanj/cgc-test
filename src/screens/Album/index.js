import React from 'react';
import {FlatList} from 'react-native';
import {View, Text} from 'react-native-ui-lib';

import {useRoute} from '@react-navigation/native';
import {useImages} from '@cgc/services/apis';
import {PhotoItem} from './PhotoItem';

export const Album = () => {
  const route = useRoute();
  const {photos, isLoading} = useImages(route.params.id);
  const keyExtractor = item => item.id;

  return (
    <View flex>
      <FlatList
        data={photos}
        renderItem={({item}) => <PhotoItem key={item.id} item={item} />}
        keyExtractor={keyExtractor}
      />
    </View>
  );
};
