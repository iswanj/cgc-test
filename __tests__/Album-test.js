import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {act} from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import {Album} from '@cgc/screens/Album';

// Silence the warning https://github.com/facebook/react-native/issues/11094#issuecomment-263240420
// jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
const mockedDispatch = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
    useRoute: () => ({
      params: {
        id: '1111',
      },
    }),
  };
});

describe('Album screen', () => {
  beforeEach(() => {
    // Alternatively, set "clearMocks" in your Jest config to "true"
    mockedDispatch.mockClear();
  });

  it('should match snapshot', async () => {
    const result = render(
      <NavigationContainer>
        <Album />
      </NavigationContainer>,
    );
    await act(async () => {
      expect(result).toMatchSnapshot();
    });
  });
});
