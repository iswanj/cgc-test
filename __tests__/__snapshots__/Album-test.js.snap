// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album screen should match snapshot 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Object {
        "flex": 1,
      },
      undefined,
      Object {},
      Object {},
      Object {},
      undefined,
    ]
  }
>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
